ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -164,6 +164,8 @@ func (r *RayClusterReconciler) eventReconcile(request ctrl.Request, event *v1.Ev
 func (r *RayClusterReconciler) rayClusterReconcile(request ctrl.Request, instance *rayiov1alpha1.RayCluster) (ctrl.Result, error) {
 	_ = r.Log.WithValues("raycluster", request.NamespacedName)
 	r.Log.Info("reconciling RayCluster", "cluster name", request.Name)
+	r.Log.Info("reconciling RayCluster", "head envs", instance.Spec.HeadGroupSpec.Envs)
```

Comment:
@Jeffwan The existing `instance.Spec.HeadGroupSpec.RayStartParams` is normal, while `instance.Spec.HeadGroupSpec.Envs` is null. I think there must be something missing, but i can't figure it out. Please take a look at it, tks. :)

Reply:
After you change types.go, you need to run `make manifest` to generate the latest CRD and create it. I think that's the reason.

Reply:
That's exactly the reason!